server/webhook.go

```diff
@@ -38,145 +45,327 @@ type Webhook struct {
+
+	case "issue_updated",
+		"issue_generic":
+		// edited summary, description, updated priority, status, etc.
+		headline = md_HeadlineFromChangeLog(w)
+		if w.showDescription {
+			extra = md_IssueDescription(w)
 		}
-	case "jira:issue_deleted":
-		if w.Issue.Fields.Resolution != nil {
-			return nil, nil
+	}
+	if headline == "" {
+		// return fmt.Sprintf("```%v```\n", w.RawJSON)
+		return ""
+	}
+
+	s += md_User(&w.User) + " " + headline + " " + md_IssueHashtags(w)
+	s += "\n"
+
+	if extra != "" {
+		s += extra
+	}
+	s += "\n"
+
+	return s
+}
+
+func mdComment(w *Webhook) string {
+	s := md_UpdateStyle
+	headline := ""
+	extra := ""
+
+	switch w.WebhookEvent {
+	case "comment_deleted":
+		headline = fmt.Sprintf("removed a comment from %v", md_IssueLongLink(w))
+
+	case "comment_updated":
+		headline = fmt.Sprintf("edited a comment in %v", md_IssueLongLink(w))
+		extra = w.Comment.Body
+
+	case "comment_created":
+		headline = fmt.Sprintf("commented on %v", md_IssueLongLink(w))
+		extra = w.Comment.Body
+
+	}
+	if headline == "" {
+		return ""
+	}
+
+	s += md_User(&w.Comment.UpdateAuthor) + " " + headline + " " + md_IssueHashtags(w)
+	s += "\n"
+	if extra != "" {
+		s += extra
+	}
+	s += "\n"
+	return s
+}
+
+func md_HeadlineFromChangeLog(w *Webhook) string {
+	for _, item := range w.ChangeLog.Items {
+		to := item.ToString
+		from := item.FromString
+		switch {
+		case item.Field == "resolution" && to == "" && from != "":
+			return fmt.Sprintf("reopened %v", md_IssueLongLink(w))
+
+		case item.Field == "resolution" && to != "" && from == "":
+			return fmt.Sprintf("resolved %v", md_IssueLongLink(w))
+
+		case item.Field == "status" && to == "In Progress":
+			return fmt.Sprintf("started working on %v", md_IssueLongLink(w))
+
+		case item.Field == "priority" && item.From > item.To:
+			return fmt.Sprintf("raised priority of %v to %v", md_IssueLongLink(w), to)
+
+		case item.Field == "priority" && item.From < item.To:
+			return fmt.Sprintf("lowered priority of %v to %v", md_IssueLongLink(w), to)
+
+		case item.Field == "summary":
+			return fmt.Sprintf("renamed %v to %v", md_IssueLink(w), md_IssueSummary(w))
+
+		case item.Field == "description":
+			w.showDescription = true
+			return fmt.Sprintf("edited description of %v", md_IssueLongLink(w))
+
+		case item.Field == "Sprint" && len(to) > 0:
+			return fmt.Sprintf("moved %v to %v", md_IssueLongLink(w), to)
+
+		case item.Field == "Rank" && len(to) > 0:
+			return fmt.Sprintf("%v %v", strings.ToLower(to), md_IssueLongLink(w))
+
+		case item.Field == "Attachment":
+			return fmt.Sprintf("%v %v", md_AddRemove(from, to, "attached", "removed attachments"), md_IssueLongLink(w))
+
+		case item.Field == "labels":
+			return fmt.Sprintf("%v %v", md_AddRemove(from, to, "added labels", "removed labels"), md_IssueLongLink(w))
+
+		default:
+			// return fmt.Sprintf("updated %v from %v to %v on %v", item.Field, from, to, md_IssueLongLink(w))
```

Comment:
Do we really want a default case here? Is the list of Fields exhaustive? Then we can add an log message.

Reply:
No we don't. This is where I was just outputting raw JSON before to see what was coming through. Will remove.